lib/samson/performance_tracer.rb

```diff
@@ -1,41 +1,22 @@
 # frozen_string_literal: true
 module Samson
   module PerformanceTracer
-    # It's used to trace the hook fire event,
-    # We can't use the samson hook to fetch the available plugins.
-    TRACER_PLUGINS = ['SamsonNewRelic', 'SamsonDatadogTracer::APM'].freeze
-
     class << self
-      def included(clazz)
-        clazz.extend ClassMethods
-      end
-
-      def trace_execution_scoped(scope_name)
-        # Tracing the scope is restricted to avoid into slow startup
-        # Refer Samson::BootCheck
-        if ['staging', 'production'].include?(Rails.env)
-          plugins = TRACER_PLUGINS.map(&:safe_constantize).compact
-          execution = using_plugins plugins, scope_name do
-            yield
-          end
-          execution.call
-        else
-          yield
-        end
+      # NOTE: this cannot be a hook since it is used from Hooks#fire
+      def handlers
+        @handlers ||= []
       end
 
-      def using_plugins(plugins, scope_name, &block)
-        plugins.inject(block) { |inner, plugin| plugin.trace_method_execution_scope(scope_name) { inner } }
+      def trace_execution_scoped(scope_name, &block)
```

Comment:
I'm unable to reproduce that only getting `Unable to verify local docker!`
also not sure where the threading would come in ... I've seen rollbar/airbrake create threads when errors happen ... is that what you saw ?

Reply:
yes, right!! rollbar/airbrake created an threads for Datadog.tracer

Reply:
Unable to verify local docker!
Extra threads: [#<Thread:0x00007feae49fb4e8@/Users/ssubramanian/Code/zendesk/samson/vendor/cache/ruby/2.5.0/gems/ddtrace-0.14.2/lib/ddtrace/workers.rb:71 sleep>]

Reply:
awesome, was able to reproduce with tracer on and docker off :)
... changed docker to not use error reporting service in dev since that already caused an issue before and is kinda pointless anyway :)
... looking good now ?

Reply:
looks good!!